xcell/cls/cl.py

```diff
@@ -261,160 +261,22 @@ def __init__(self, data, tr1, tr2):
 
         self.outdir = self.get_outdir('fiducial')
         os.makedirs(self.outdir, exist_ok=True)
-        self.cosmo = self.get_cosmo_ccl()
-        self.hm_par = self._get_halomodel_params()
+        self.th = Theory(self.data.data)
         self._ccl_tr1 = None
         self._ccl_tr2 = None
 
-    def _get_halomodel_params(self):
-        if 'halo_model' not in self.data.data['cov']['fiducial']:
-            self.data.data['cov']['fiducial']['halo_model'] = {}
-        hmp = self.data.data['cov']['fiducial']['halo_model']
-
-        if 'mass_def' not in hmp:
-            md = ccl.halos.MassDef200m()
-        else:
-            mds = hmp['mass_def']
-            Delta = float(mds[:-1])
-            if mds[-1] == 'm':
-                rho_type = 'matter'
-            elif mds[-1] == 'c':
-                rho_type = 'critical'
-            else:
-                raise ValueError("Unknown density type %s" % (mds[-1]))
-            md = ccl.halos.MassDef(Delta, rho_type)
-
-        mfc = ccl.halos.mass_function_from_name(hmp.get('mass_function',
-                                                        'Tinker10'))
-        mf = mfc(self.cosmo, mass_def=md)
-
-        hbc = ccl.halos.halo_bias_from_name(hmp.get('halo_bias',
-                                                    'Tinker10'))
-        hb = hbc(self.cosmo, mass_def=md)
-
-        # We also need an NFW profile to handle certain cases
-        cmc = ccl.halos.concentration_from_name(hmp.get('concentration',
-                                                        'Duffy08'))
-        cm = cmc(mdef=md)
-        pNFW = ccl.halos.HaloProfileNFW(cm)
-
-        # Halo model calculator
-        hmc = ccl.halos.HMCalculator(self.cosmo, mf, hb, md)
-
-        # Transition smoothing
-        alpha = hmp.get('alpha_HMCODE', 0.7)
-        # Small-k damping
-        klow = hmp.get('k_suppress', 0.01)
-
-        return {'mass_def': md,
-                'mass_func': mf,
-                'halo_bias': hb,
-                'cM': cm,
-                'prof_NFW': pNFW,
-                'calculator': hmc,
-                'alpha': (lambda a: alpha),
-                'k_suppress': (lambda a: klow)}
-
-    def get_cosmo_ccl(self):
-        fiducial = self.data.data['cov']['fiducial']
-        cosmo = ccl.Cosmology(**fiducial['cosmo'])
-        return cosmo
-
     def get_tracers_ccl(self):
```

Comment:
Why don't you move this to `Theory` as well?

Reply:
Because this is specific of power spectra, which involve a pair of tracers. There'll be something like that for covariances too.